### ISSUE TITLE
False error when determining type of `decode` function of `BorshSerialize` trait

### ISSUE THREAD

**Issue by pmnoxx** [intellij-rust/intellij-rust#8415]
I see that the plugin reports an error for a correct code.

```
mismatches types [E308]
expected `&HandshakeAutoDes`, for `&mut &[u8]`.
```

For line
```rust
HandshakeAutoDes::deserialize(buf).map(Into::into)
```

See https://github.com/near/nearcore/pull/6076

## Environment

* **IntelliJ Rust plugin version:**
* 0.4.162.4321-213
* **Rust toolchain version:**                                            
* cargo 1.57.0 (b2e52d7ca 2021-10-21)
* **IDE name and version:**
* CLion 2021.3.2
* **Operating system:**
* Ubuntu 20.04

## Problem description

As  seen in 
See https://github.com/near/nearcore/pull/6076

This produces the error:
```rust
HandshakeAutoDes::deserialize(buf).map(Into::into)
```
While this behaves properly:
```rust
<HandshakeAutoDes as BorshDeserialize>::deserialize(buf).map(Into::into)
```

## Steps to reproduce

This is an issue for our repository. Here is the PR that fixes the issue for our repo.

See https://github.com/near/nearcore/pull/6076

This can be reproduces by loading the change for the commit before this PR.

**Comment by Undin** [comment]
As I can see, the corresponding error is provided by the plugin only when procedural macro expansion is disabled (it's disabled by default). It's expected since the plugin doesn't know about `impl` generated by `BorshDeserialize` proc macro invocation.
Try to enable enable `org.rust.cargo.evaluate.build.scripts` and `org.rust.macros.proc` [experimental](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features) features to turn on procedural macro expansion

See #6908 for more details

**Comment by Undin** [comment]
Let's keep it open since procedural macro expansion is not enabled by default
